### ISSUE TITLE
Call to undefined local scope when querying the model

### ISSUE THREAD

**Issue by tsushant** [nunomaduro/larastan#585]
- Larastan Version: 0.5.2
- `--level`: 5

### Description

The error is while calling the scope method **search** when querying the model

`Call to an undefined method Illuminate\Database\Eloquent\Builder::search()`

### Laravel code where the issue was found

```php
public function scopeSearch($query, $search)
{
    if ($search) {
        return $query->where('name', 'like', '%'.$search.'%');
    }
    return $query;
}
```
`$this->model->search(request('search'))->paginate();`

_Originally posted by @tsushant in https://github.com/nunomaduro/larastan/issues/28#issuecomment-633566694_

**Comment by szepeviktor** [comment]
Please try to upgrade Larastan.

**Comment by canvural** [comment]
Hi,

Can you also share what is the type of `$this->model`?

**Comment by tsushant** [comment]
Hi,
```
abstract class EloquentBaseRepository implements BaseRepository
{
    /**
     * The repository model.
     *
     * @var \Illuminate\Database\Eloquent\Model
     */
    protected $model;

    /**
     * Return a listing of the given model resource.
     *
     * @param  array $columns
     * @return \Illuminate\Support\Collection
     */
    public function all($columns = ['*']): Collection
    {
        return $this->model->get($columns);
    }
}
```
I updated to **0.5.8** the above error is gone but this error appeared `Call to an undefined method Illuminate\Database\Eloquent\Model::get()` which was not showing before.

Thanks :)

**Comment by canvural** [comment]
Ah, I see. It has the abstract class `Model` as typehint. I think this is solved by #565 You can try the latest `master` and check if it solves your issue. And the next version will be released soon :+1:

**Comment by canvural** [comment]
Closing this because it was fixed. Feel free to open a new issue if the issue still persists.
